aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/SynchronousStorageAdapter.java

```diff
@@ -237,6 +239,28 @@ public void terminate() throws DataStoreException {
         );
     }
 
+    /**
+     * Delete every model of given type that matches predicate.
+     * @param modelType Model type to delete
+     * @param predicate Filter to apply for deletion
+     * @param <T> Type of model being deleted
+     * @throws DataStoreException On any failure to delete the models
+     */
+    public <T extends Model> void delete(@NonNull Class<T> modelType, @NonNull QueryPredicate predicate)
+            throws DataStoreException {
+        Await.result(
+                operationTimeoutMs,
+                (Consumer<Object> onResult, Consumer<DataStoreException> onError) ->
```

Comment:
Oh, interesting. I usually provide the type bounds in `Await.<Here, here>`, but it looks like they are inferred if you add them to the lambda, too, eh.